src/utils/updateLegacyIssuers.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  DomainHierarchy,
+  DomainReader,
+  DomainTransactionFactory,
+  IRoleDefinition,
+  ResolverContractType,
+} from '@energyweb/iam-contracts';
+import { providers } from 'ethers';
+import { Methods } from '@ew-did-registry/did';
+import { Signer } from 'ethers';
+import { isValidDID } from './did';
+import { ENSRegistry__factory } from '../../ethers/factories/ENSRegistry__factory';
+import { labelhash, namehash } from './ensHash';
+import { ChainId, chainConfigs } from '../config';
+
+const { JsonRpcProvider } = providers;
+
+/**
+ * @description - Checks that role issuers of all roles under `rootDomain` contains method-specific-id and adds it if missing
+ * `signer` must own `rootDomain` on `targetChain`
+ */
+export const updateLegacyIssuers = async (
+  rootDomain: string,
+  signer: Signer,
+  chainId: ChainId,
+  dryRun = true
+) => {
+  const {
+    transactionFactory,
+    chainName,
+    provider,
+    domainHierarchy,
+    domainReader,
+    ensRegistry,
+    ensResolverV2Address,
+  } = await initDomains(signer, chainId);
+
+  let updated = new Array<any>();
+  const update = async (domain: string) => {
+    const domainHash = namehash(domain);
+    const def = await domainReader.read({ node: domainHash });
+    let subnodes = await domainHierarchy.getSubdomainsUsingResolver({
+      domain,
+      mode: 'FIRSTLEVEL',
+    });
+    if (DomainReader.isOrgDefinition(def) || DomainReader.isAppDefinition(def)) {
+      subnodes = subnodes.concat(
+        await domainHierarchy.getSubdomainsUsingResolver({
+          domain: `roles.${domain}`,
+          mode: 'FIRSTLEVEL',
+        })
+      );
+    }
+    if (DomainReader.isOrgDefinition(def)) {
+      subnodes = subnodes.concat(
+        await domainHierarchy.getSubdomainsUsingResolver({
+          domain: `apps.${domain}`,
+          mode: 'FIRSTLEVEL',
+        })
+      );
+    }
+
+    const resolver = await ensRegistry.resolver(domainHash);
+    if (resolver !== ensResolverV2Address) {
+      if (!dryRun) {
+        await (await ensRegistry.setResolver(domain, ensResolverV2Address)).wait();
+      }
+      let updateDomainTx;
+      if (DomainReader.isRoleDefinition(def)) {
+        const {
+          issuer: { did },
+          version,
+        } = def;
+        const updatedDef: IRoleDefinition = {
+          ...def,
+          issuer: {
+            ...def.issuer,
+            did: did?.map((d) =>
+              isValidDID(d) ? d : `did:${Methods.Erc1056}:${chainName}:${d.split(':').slice(-1)}`
+            ),
+          },
+          version: parseInt(version.toString(), 10),
+        };
+        updated.push({ domain, legacyDef: def, updatedDef });
+        updateDomainTx = transactionFactory.newRole({ domain, roleDefinition: updatedDef });
```

Comment:
@JGiter Is a tx to the ENS Registry also required to update resolver that the domain is using? I don't think that this is currently being covered. https://github.com/energywebfoundation/iam-client-lib/blob/eae6b169f5c9e960655cd38a6f8d414297bd55d1/src/modules/domains/domains.service.ts#L908

Reply:
Oh nvm, sorry, I see that you are doing it already with `await ensRegistry.setResolver(domain, ensResolverV2Address)`